strawberry/auto.py

```diff
@@ -1,18 +1,50 @@
-from typing import Any, ClassVar, Optional
+from typing import Any, Optional, Union, cast
 
-from typing_extensions import Annotated, Final
+from typing_extensions import Annotated, Final, get_args, get_origin
 
+from strawberry.type import StrawberryType
 
-class StrawberryAuto:
-    _instance: ClassVar[Optional["StrawberryAuto"]] = None
+from .annotation import StrawberryAnnotation
 
-    def __new__(cls, *args, **kwargs):
+
+class StrawberryAutoMeta(type):
+    def __init__(self, *args, **kwargs):
+        self._instance: Optional["StrawberryAuto"] = None
```

Comment:
Nit: can we `from __future__ import annotations` here so we don't have to use the quotes?

Reply:
Also, I would recommend adding a docstring here indicating the metaclass is used to make `StrawberryAuto` a singleton just so it's a little more explicit